### PR TITLE
Fix: missing parens around labeled record pattern arg

### PR DESCRIPTION
**Pull request by jberdine**

### CODE REVIEW

File: src/Ast.ml [hunk 1]
```diff
@@ -1329,11 +1329,14 @@ end = struct
       , inner ) -> (
       match inner with
       | Ppat_construct ({txt= Lident "::"}, _) -> true
-      | Ppat_construct _ | Ppat_variant _ -> false
+      | Ppat_construct _ | Ppat_record _ | Ppat_variant _ -> false
```
[the remark this answers is not in the document]
Comment (jberdine) [reply src/Ast.ml]: Without this, then the other changes in this patch cause every record in a sugared list to be needlessly parenthesized. Added a test to check that extra parens are removed.

File: src/Ast.ml [hunk 2]
```diff
@@ -1329,11 +1329,12 @@ end = struct
...
       | _ -> true )
     | ( Pat {ppat_desc= Ppat_construct _}
       , Ppat_construct ({txt= Lident "::"}, _) ) ->
         true
+    | Exp {pexp_desc= Pexp_fun (Optional _, _, _, _)}, Ppat_record _ -> true
```
Comment (jberdine) [inline src/Ast.ml]: Entertainingly, this is only needed for Optional, but not Labelled args.

File: test/passing/record.ml [hunk 3]
```diff
@@ -27,3 +27,5 @@ let _ = {(x : x) with e1; e2}
 let _ = {(x :> x) with e1; e2}
 
 let _ = {(x#x) with e1; e2}
+
+let f ?l:({f; g}) = e
```
Comment (hhugo) [inline test/passing/record.ml]: can you add a test with a non optional label

Comment (jberdine) [reply test/passing/record.ml]: done

**Pull request merged**